newIDE/app/src/EventsSheet/ParameterFields/LeaderboardIdField.js

```diff
@@ -0,0 +1,147 @@
+
+const getLeaderboardNameByValue = (
+  leaderboards: ?Array<Leaderboard>,
+  value: string
+): string => {
+  if (!leaderboards) return value;
+  const quotedIdLeaderboards = leaderboards.map(leaderboard => ({
+    id: `"${leaderboard.id}"`,
+    name: leaderboard.name,
+  }));
+  const leaderboard = quotedIdLeaderboards.find(
+    leaderboard => leaderboard.id === value
+  );
+  return leaderboard ? leaderboard.name : value;
+};
+
+export function LeaderboardIdField(props: ParameterFieldProps) {
+  const { leaderboards } = React.useContext(LeaderboardContext);
+  const [isAdminOpen, setIsAdminOpen] = React.useState(false);
+
+  const isCurrentValueInLeaderboardList =
+    leaderboards &&
+    !!leaderboards.find(leaderboard => `"${leaderboard.id}"` === props.value);
+
+  const [isTextInput, setIsTextInput] = React.useState(
+    !leaderboards || (!!props.value && !isCurrentValueInLeaderboardList)
+  );
+
+  const onChangeSelectValue = (event, value) => {
+    props.onChange(event.target.value);
+  };
+
+  const onChangeTextValue = (value: string) => {
+    props.onChange(value);
+  };
+
+  const fieldLabel = props.parameterMetadata
+    ? props.parameterMetadata.getDescription()
+    : undefined;
+
+  return (
+    <>
+      <TextFieldWithButtonLayout
+        renderTextField={() =>
+          !isTextInput ? (
+            <SelectField
+              value={props.value}
+              onChange={onChangeSelectValue}
+              margin={props.isInline ? 'none' : 'dense'}
+              fullWidth
+              floatingLabelText={fieldLabel}
+              hintText={t`Choose a leaderboard`}
+            >
+              {leaderboards && !!leaderboards.length
+                ? leaderboards.map(leaderboard => (
+                    <SelectOption
+                      key={leaderboard.id}
+                      value={`"${leaderboard.id}"`}
+                      primaryText={`${leaderboard.name} ${
+                        leaderboard.id
+                          ? `(${leaderboard.id.substring(0, 8)})`
+                          : ''
+                      }`}
+                    />
+                  ))
+                : null}
+            </SelectField>
+          ) : (
+            <GenericExpressionField
+              expressionType="string"
+              {...props}
+              onChange={onChangeTextValue}
+              onExtractAdditionalErrors={(
+                currentExpression: string,
+                currentExpressionNode: gdExpressionNode
+              ) => {
+                if (!leaderboards)
+                  return `Unable to fetch leaderboards as you are offline.`;
+              }}
+            />
+          )
+        }
+        renderButton={style => (
+          <>
+            <RaisedButton
+              icon={<OpenInNew />}
+              primary
+              style={style}
+              onClick={() => setIsAdminOpen(true)}
+            />
+            <LargeSpacer />
+            <Toggle
```

Comment:
Is it possible to have something that explains what this toggle does? A tooltip or a title? or an icon `Info`?

Reply:
And it would explain why it is disabled when leaderboards could not have been fetched